File: src/main/java/easy/MaxIceCream.java
package easy;

import java.util.Arrays;

public class MaxIceCream {
    public int solve(int[] costs, int coins) {
        int cones = 0;
        Arrays.sort(costs);
        for (int i : costs) {
            if (coins >= i) {
                coins -= i;
                cones++;
            }
        }
        return cones;
    }
}

File: src/main/php/Easy/MaxIceCream.php
<?php

declare(strict_types=1);

namespace Src\Easy;

class MaxIceCream
{
    public function solve(array $costs, int $coins): int
    {
        $cones = 0;
        $remaining = $coins;
        sort($costs);
        foreach ($costs as $cost) {
            if ($cost <= $remaining) {
                $remaining -= $cost;
                $cones++;
            }
        }

        return $cones;
    }
}
